diff --git a/include/curl/system.h b/include/curl/system.h
index 867af6141..e2cd514c0 100644
--- a/include/curl/system.h
+++ b/include/curl/system.h
@@ -421,7 +421,11 @@
 #ifdef CURL_PULL_WS2TCPIP_H
 #  include <winsock2.h>
 #  include <windows.h>
+#if _WIN32_WINNT >= 0x0501
 #  include <ws2tcpip.h>
+#else
+#  include <wspiapi.h>
+#endif
 #endif
 
 /* CURL_PULL_SYS_TYPES_H is defined above when inclusion of header file  */
diff --git a/src/tool_doswin.c b/src/tool_doswin.c
index b7df3e615..69f51d046 100644
--- a/src/tool_doswin.c
+++ b/src/tool_doswin.c
@@ -655,6 +655,9 @@ struct curl_slist *GetLoadedModulePaths(void)
 
   mod.dwSize = sizeof(MODULEENTRY32);
 
+  /* [marius]: Toolhelp functions are available in W2K+ */
+#if _WIN32_WINNT >= _WIN32_WINNT_WIN2K
+
   do {
     hnd = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, 0);
   } while(hnd == INVALID_HANDLE_VALUE && GetLastError() == ERROR_BAD_LENGTH);
@@ -694,6 +697,8 @@ error:
 cleanup:
   if(hnd != INVALID_HANDLE_VALUE)
     CloseHandle(hnd);
+
+#endif
   return slist;
 }
 
