diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 1d71e1490..edc68db09 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -39,9 +39,9 @@ list(APPEND HHEADERS
   ${CMAKE_CURRENT_BINARY_DIR}/curl_config.h
   )
 
-if(MSVC)
+# if(MSVC) # Marius: Version info in mingw builds
   list(APPEND CSOURCES libcurl.rc)
-endif()
+# endif()
 
 # SET(CSOURCES
 # #  memdebug.c -not used
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 26e3cfe1e..25dde8957 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -57,9 +57,9 @@ endif()
 transform_makefile_inc("Makefile.inc" "${CMAKE_CURRENT_BINARY_DIR}/Makefile.inc.cmake")
 include(${CMAKE_CURRENT_BINARY_DIR}/Makefile.inc.cmake)
 
-if(MSVC)
+# if(MSVC) # Marius: Version info in mingw builds
   list(APPEND CURL_FILES curl.rc)
-endif()
+# endif()
 
 # CURL_FILES comes from Makefile.inc
 add_executable(
