diff --git a/library/platform.c b/library/platform.c
index 575615954..4e2207bfa 100644
--- a/library/platform.c
+++ b/library/platform.c
@@ -137,7 +137,7 @@ int mbedtls_platform_win32_vsnprintf( char *s, size_t n, const char *fmt, va_lis
     if( s == NULL || n == 0 || fmt == NULL )
         return( -1 );
 
-#if defined(_TRUNCATE)
+#if defined(_TRUNCATE) && !defined(__MINGW32__)
     ret = vsnprintf_s( s, n, _TRUNCATE, fmt, arg );
 #else
     ret = vsnprintf( s, n, fmt, arg );
