 library/net_sockets.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/library/net_sockets.c b/library/net_sockets.c
index 7b4a423c..35be459e 100644
--- a/library/net_sockets.c
+++ b/library/net_sockets.c
@@ -47,12 +47,15 @@
 
 #define IS_EINTR( ret ) ( ( ret ) == WSAEINTR )
 
-#ifdef _WIN32_WINNT
-#undef _WIN32_WINNT
+#ifndef _WIN32_WINNT
+    #define _WIN32_WINNT 0x0501
 #endif
 /* Enables getaddrinfo() & Co */
-#define _WIN32_WINNT 0x0501
-#include <ws2tcpip.h>
+#if _WIN32_WINNT >= 0x0501
+    #include <ws2tcpip.h>
+#else
+    #include <wspiapi.h>
+#endif
 
 #include <winsock2.h>
 #include <windows.h>
