 library/net_sockets.c | 11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/library/net_sockets.c b/library/net_sockets.c
index cc06cbf..9bbd263 100644
--- a/library/net_sockets.c
+++ b/library/net_sockets.c
@@ -45,12 +45,15 @@
 #if (defined(_WIN32) || defined(_WIN32_WCE)) && !defined(EFIX64) && \
     !defined(EFI32)
 
-#ifdef _WIN32_WINNT
-#undef _WIN32_WINNT
+#ifndef _WIN32_WINNT
+    #define _WIN32_WINNT 0x0501
 #endif
 /* Enables getaddrinfo() & Co */
-#define _WIN32_WINNT 0x0501
-#include <ws2tcpip.h>
+#if _WIN32_WINNT >= 0x0501
+    #include <ws2tcpip.h>
+#else
+    #include <wspiapi.h>
+#endif
 
 #include <winsock2.h>
 #include <windows.h>
