 library/net_sockets.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/library/net_sockets.c b/library/net_sockets.c
index 816b1303d..d830ab3af 100644
--- a/library/net_sockets.c
+++ b/library/net_sockets.c
@@ -53,14 +53,16 @@
 
 #define IS_EINTR( ret ) ( ( ret ) == WSAEINTR )
 
-#if !defined(_WIN32_WINNT) || (_WIN32_WINNT < 0x0501)
-#undef _WIN32_WINNT
+#if !defined(_WIN32_WINNT)
+    #define _WIN32_WINNT 0x0501
+#endif
 /* Enables getaddrinfo() & Co */
-#define _WIN32_WINNT 0x0501
+#if _WIN32_WINNT >= 0x0501
+    #include <ws2tcpip.h>
+#else
+    #include <wspiapi.h>
 #endif
 
-#include <ws2tcpip.h>
-
 #include <winsock2.h>
 #include <windows.h>
 
