 lib/Makefile.m32 | 1 +
 src/Makefile.m32 | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lib/Makefile.m32 b/lib/Makefile.m32
index 522dbd1..a3bc7f7 100644
--- a/lib/Makefile.m32
+++ b/lib/Makefile.m32
@@ -310,6 +310,7 @@ ifndef USE_LDAP_OPENLDAP
 endif
 endif
 DLL_LIBS += -lws2_32
+DLL_LIBS += $(CURL_LDFLAG_EXTRAS2)
 
 # Makefile.inc provides the CSOURCES and HHEADERS defines
 include Makefile.inc
diff --git a/src/Makefile.m32 b/src/Makefile.m32
index 40852e5..10b281d 100644
--- a/src/Makefile.m32
+++ b/src/Makefile.m32
@@ -331,6 +331,7 @@ curl_LDADD += -lwldap32
 endif
 endif
 curl_LDADD += -lws2_32
+curl_LDADD += $(CURL_LDFLAG_EXTRAS2)
 
 # Makefile.inc provides the CSOURCES and HHEADERS defines
 include Makefile.inc
