 lib/Makefile.m32   | 1 +
 src/Makefile.m32   | 1 +
 2 files changed, 2 insertions(+), 0 deletion(-)

diff --git a/lib/Makefile.m32 b/lib/Makefile.m32
index 22efbdee0..39fadcf63 100644
--- a/lib/Makefile.m32
+++ b/lib/Makefile.m32
@@ -309,6 +309,7 @@ ifndef USE_LDAP_OPENLDAP
 endif
 endif
 DLL_LIBS += -lws2_32
+DLL_LIBS += $(CURL_LDFLAG_EXTRAS2)
 
 # Makefile.inc provides the CSOURCES and HHEADERS defines
 include Makefile.inc
diff --git a/src/Makefile.m32 b/src/Makefile.m32
index e55427a5e..3d9a04f62 100644
--- a/src/Makefile.m32
+++ b/src/Makefile.m32
@@ -331,6 +331,7 @@ curl_LDADD += -lwldap32
 endif
 endif
 curl_LDADD += -lws2_32
+curl_LDADD += $(CURL_LDFLAG_EXTRAS2)
 
 # Makefile.inc provides the CSOURCES and HHEADERS defines
 include Makefile.inc
